monitor/protocolLevel.js

```diff
@@ -0,0 +1,491 @@
+            );
+        }
+
+        if (
+            dataFields.includes("cumulativeDepositUSD") &&
+            !(
+                parseFloat(data.cumulativeDepositUSD) >=
+                parseFloat(data.totalDepositBalanceUSD)
+            )
+        ) {
+            issuesArrays.cumulativeDepo.push(
+                deploymentName +
+                "//" +
+                parseFloat(data.cumulativeDepositUSD).toFixed(2)
+            );
+        }
+
+        if (
+            dataFields.includes("totalBorrowBalanceUSD") &&
+            !(
+                parseFloat(data.totalBorrowBalanceUSD) <=
+                parseFloat(data.totalDepositBalanceUSD)
+            )
+        ) {
+            issuesArrays.totalBorrowBal.push(
+                deploymentName +
+                "//" +
+                parseFloat(data.totalBorrowBalanceUSD).toFixed(2)
+            );
+        }
+
+        if (
+            dataFields.includes("cumulativeLiquidateUSD") &&
+            !(
+                parseFloat(data.cumulativeLiquidateUSD) <=
+                parseFloat(data.cumulativeBorrowUSD)
+            )
+        ) {
+            issuesArrays.cumulativeLiquidate.push(
+                deploymentName +
+                "//" +
+                parseFloat(data.cumulativeLiquidateUSD).toFixed(2)
+            );
+        }
+
+        deployments[deploymentName].protocolErrors = issuesArrays;
+    });
+
+    const depoHolder = {};
+    Object.keys(deployments).forEach((x) => {
+        depoHolder[x] = { ...deployments[x] };
+        const key = depoHolder[x].protocolErrors;
+        Object.keys(key).forEach((y) => {
+            if (key[y].length !== 0) {
+                depoHolder[x].protocolErrors[y] = key[y];
+            }
+        });
+    });
+    return depoHolder;
+}
+
+export const alertProtocolErrors = async (discordMessages, deployments) => {
+    const protocolErrorMessageObjs = discordMessages.filter(x => x.content.includes("**PROTOCOL ERRORS:**"));
+    const protocolErrorMsgs = protocolErrorMessageObjs.map(msgObj => {
+        return msgObj.content.split("TYPE:")[1];
+    })
+
+    const errorsOnType = {
+        tvlRange: [],
+        cumulativeSupplySideRev: [],
+        cumulativeProtocolSideRev: [],
+        cumulativeTotalRev: [],
+        cumulativeVol: [],
+        cumulativeUniqueUsers: [],
+        totalPoolCount: [],
+        cumulativeUniqueDepos: [],
+        cumulativeUniqueBorrowers: [],
+        cumulativeUniqueLiquidators: [],
+        cumulativeUniqueLiquidatees: [],
+        openPositionCount: [],
+        cumulativePositionCount: [],
+        totalDepoBal: [],
+        cumulativeDepo: [],
+        totalBorrowBal: [],
+        cumulativeLiquidate: [],
+    };
+    protocolErrorMsgs.forEach(msgObj => {
+        const splitMsg = msgObj.split("LIST:");
+        const type = splitMsg[0].split('\n')[0].trim();
+        errorsOnType[type] = [...errorsOnType[type], ...splitMsg[1]].join('');
+    })
+
+    const currentErrors = {
```

Comment:
what's the difference between this and errorsOnType?

Reply:
errorsOnType contains the issues detected and alerted for in the prior week. currentErrors are the issues detected that are not contained in errorsOnType and need to be alerted for. Ill rename these to something that better describes their roles